### ISSUE TITLE
Improve diffing of Union types

### ISSUE THREAD

**Issue by harrysarson** [rtfeldman/node-test-runner#263]
Hi, I have come accross a difficulty when testingUnion types.

I have a test along the lines of:

```elm
test "union type diff" <|
    \() ->
        Err { description = "An error", context = "Explanation of what went so wrong" }
            |> Expect.equal (Ok "Success.")
```

This results in the following output:
![image](https://user-images.githubusercontent.com/16308754/42658789-451a32c6-861e-11e8-8aad-784276caf29b.png)

The diff is not very helpful, futhermore the attempts to highlight the differing characters make it hard to read the error message.

I was wondering if it would be possible to detect the case where different union types are being compared and format this case differently. 

Sorry if I have opened this issue in the wrong repo.
Cheers.

**Comment by mgold** [comment]
I don't think the text being presented is that bad, but wow is the text highlighting bad. Reproduced.

**Comment by rtfeldman** [comment]
Fix published in `0.19.0-rev5`
